app/components/blacklight/hierarchy/facet_field_component.rb

```diff
@@ -33,6 +31,10 @@ def qfacet_selected?
         config = helpers.facet_configuration_for_field(field_name)
         helpers.search_state.filter(config).include?(item.qvalue)
       end
+
+      def ul_id
+        @ul_id ||= "b-h-#{SecureRandom.alphanumeric(10)}"
```

Comment:
Where was this called more than one time?

Reply:
`ul_id` is used in L2 and L12 of app/components/blacklight/hierarchy/facet_field_component.html.erb

Previously, SecureRandom.alphanumeric(10) was called in facet_field_component.html.erb and facet_field_component.rb